### ISSUE TITLE
Integration builds fail with arm64 base image

### ISSUE THREAD

**Issue by christophd** [apache/camel-k#5238]
### What happened?

When Camel K is run on arm64 architecture the user needs to use a different base image for integrations (e.g. `--base-image=arm64v8/eclipse-temurin:17`).

When the Camel K operator builds integration kit image (strategy=routine) for Integrations the build fails in the operator with Jib not being able to use this arm64 base image.

A workaround is to use a a specific image digest on the base image (e.g. `--base-image=arm64v8/eclipse-temurin:17@sha256:321ac6af97a11d5aa5a580d1635ab6f945582b00eea1fe672f0cbb437b4293c8`). With this all Integration builds succeed as expected.

### Steps to reproduce

1. Install Camel K on arm64 arch (`kamel install --operator-image=apache/camel-k:2.3.0-SNAPSHOT-arm64 --base-image=arm64v8/eclipse-temurin:17`)
2. Run an Integration and review build log output in operator logs - see the build failing

### Relevant log output

```shell
{"level":"error","ts":"2024-03-11T08:26:16Z","logger":"camel-k.maven.build","msg":"arm64v8/eclipse-temurin:17 is a manifest list, but the list does not contain an image for architecture=amd64, os=linux. If your intention was to specify a platform for your image, see https://github.com/GoogleContainerTools/jib/blob/master/docs/faq.md#how-do-i-specify-a-platform-in-the-manifest-list-or-oci-index-of-a-base-image","stacktrace":"github.com/apache/camel-k/v2/pkg/util/log.Logger.Error\n\tgithub.com/apache/camel-k/v2/pkg/util/log/log.go:81\ngithub.com/apache/camel-k/v2/pkg/util/maven.normalizeLog\n\tgithub.com/apache/camel-k/v2/pkg/util/maven/maven_log.go:83\ngithub.com/apache/camel-k/v2/pkg/util/maven.MavenLogHandler\n\tgithub.com/apache/camel-k/v2/pkg/util/maven/maven_log.go:54\ngithub.com/apache/camel-k/v2/pkg/util.scan\n\tgithub.com/apache/camel-k/v2/pkg/util/command.go:76\ngithub.com/apache/camel-k/v2/pkg/util.RunAndLog.func1\n\tgithub.com/apache/camel-k/v2/pkg/util/command.go:55\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.6.0/errgroup/errgroup.go:78"}
```

### Camel K version

2.3.0-SNAPSHOT

**Comment by squakez** [comment]
This is already documented: https://camel.apache.org/camel-k/2.2.x/installation/advanced/multi-architecture.html

**Comment by christophd** [comment]
Thanks for pointing to the docs - I was not aware of this.

But having to use the sha256 digest prevents us from getting the very latest updates from an image stream `arm64v8/eclipse-temurin:17`, right? When there is a new update pushed to `arm64v8/eclipse-temurin:17` we need to manually adjust the base image digest on the base image.

Do we know why we need to use the digest in the base image here? Maybe @gansheer?

**Comment by tdiesler** [comment]
You can assign this to me - I wanted to look at this base image issue anyway

**Comment by tdiesler** [comment]
Depends on #5200

**Comment by tdiesler** [comment]
PR: https://github.com/apache/camel-k/pull/5246

**Comment by tdiesler** [comment]
During build time, we extract the base image digest from the manifest list, like this ...

```
ARM64_DIGEST := $(shell docker manifest inspect eclipse-temurin:17 | jq -r '.manifests[] | select(.platform.architecture == "arm64") | .digest')
```

We use that base image, if no other was explicitly defined

```
$ kamel install --olm=false --operator-image apache/camel-k:2.3.0-SNAPSHOT

Using linux/arm64 base image: eclipse-temurin:17@sha256:321ac6af97a11d5aa5a580d1635ab6f945582b00eea1fe672f0cbb437b4293c8
Camel K installed in namespace default 
```

**Comment by tdiesler** [comment]
Slack: https://redhat-internal.slack.com/archives/C06DH6BJA80/p1710406352615669

**Comment by tdiesler** [comment]
In this second attempt, we simply tell Jib to consider multiple platforms for the FROM image

```
-Djib.from.platforms=linux/amd64,linux/arm64 -Djib.from.image=eclipse-temurin:17
```

I also upgraded to jib-3.4.1

**Comment by gansheer** [comment]
> In this second attempt, we simply tell Jib to consider multiple platforms for the FROM image
> 
> ```
> -Djib.from.platforms=linux/amd64,linux/arm64 -Djib.from.image=eclipse-temurin:17
> ```
> 
> I also upgraded to jib-3.4.1

That's nice !

I guess that means by default we will create images compatible with both platforms.

**Comment by tdiesler** [comment]
> I guess that means by default we will create images compatible with both platforms.

No sure that this is actually true. For now I assume that the integration image (singular) is built to run on the platform that invokes the build. Whether and how the build output is also portable to other archs would be another issue, with lots of testing and documentation to be done

In short, this PR is a fix to the issue at hand - no (positive) side effects implied

**Comment by squakez** [comment]
`docker.io/apache/camel-k:2.3.0-SNAPSHOT-arm64` is no longer valid. I think you need to install normal `docker.io/apache/camel-k:2.3.0-SNAPSHOT`. The cluster will pick the correct platform from the manifest. We amended the previous documentation https://camel.apache.org/camel-k/next/installation/advanced/multi-architecture.html

**Comment by squakez** [comment]
Also, mind that when you `make images` you should not run `--operator-image docker.io/apache...`, otherwise you'd be using the image published in docker instead of your local own.

**Comment by christophd** [comment]
> docker.io/apache/camel-k:2.3.0-SNAPSHOT-arm64 is no longer valid

why is `make images` building this image then?

> Also, mind that when you make images you should not run --operator-image docker.io/apache..., otherwise you'd be using the image published in docker instead of your local own.

I have switched to Podman recently and this is how the image gets pushed to the Minikube registry using the `docker.io/` prefix. I need to use the prefix but I guess this is MacOS specific when using the Minikube VM

**Comment by christophd** [comment]
`docker.io/apache/camel-k:2.3.0-SNAPSHOT-arm64` image works good for me. This is not the issue. the issue is that the integration base image for `arm64` architecture is not working with jib build in the operator

**Comment by squakez** [comment]
Okey, I though you were using the old nightly. In general, given we're publishing the manifest now, I think the usage of suffix is no longer required as the decision is going to be delegated to the cluster.

**Comment by christophd** [comment]
In the error I can see `Failed to create, open, or parse global Jib config file` as it expects a file in `~/.config/google-cloud-tools-java/jib/config.json`. Jib will create that file if not present but I think the process lacks write permissions for that path.

We could add `XDG_CONFIG_HOME` env var and point this to a writable directory within the Docker image.
